packages/devtools_app/lib/src/charts/flame_chart.dart

```diff
@@ -415,8 +416,7 @@ abstract class FlameChartState<T extends FlameChart, V> extends State<T>
       // to call this that guarantees the scroll controller offsets will be
       // updated for the new zoom level and layout size
       // https://github.com/flutter/devtools/issues/2012.
-      horizontalController.jumpTo(
-          newScrollOffset.clamp(FlameChart.minScrollOffset, maxScrollOffset));
+      scrollToX(newScrollOffset, jump: true);
```

Comment:
scrolllToX doesn't seem safe to call after a zoom as the scrollbar will still have the old scroll extent for a frame if I remember correctly. Perhaps we need a custom scroll controller class that lets us update its extent immediately based on the new size to avoid all these annoying edge cases?

Reply:
This change is a no-op to use the helper method instead of calling jumpTo and manually clamping the offset. We do have to scroll though in order to keep the mouse centered on the position of the scroll.

I'd have to dig in a little deeper to remember the timing of everything, but that sounds reasonable to do in a future improvement.